python/src/aiconfig/default_parsers/openai_vision.py

```diff
@@ -0,0 +1,427 @@
+            )
+
+        # Find first system prompt. Every prompt in the config will bet set to use this system prompt.
+        system_prompt = None
+        for message in conversation_data["messages"]:
+            if "role" in message and message["role"] == "system":
+                # Note: system prompt for openai represented will be an object with content and role attributes {content: str, role: str}
+                system_prompt = message
+                break
+
+        # Get the global settings for the model
+        model_name = (
+            conversation_data["model"]
+            if "model" in conversation_data
+            else self.id()
+        )
+
+        model_metadata = ai_config.get_model_metadata(
+            conversation_data, model_name
+        )
+        # Remove messages array from model metadata. Handled separately
+        if model_metadata.settings:
+            model_metadata.settings.pop("messages", None)
+
+        # Add in system prompt
+        if system_prompt:
+            if not model_metadata.settings:
+                model_metadata.settings = {}
+            model_metadata.settings["system_prompt"] = system_prompt
+
+        i = 0
+        while i < len(conversation_data["messages"]):
+            message = conversation_data["messages"][i]
+            role = message["role"]
+            if role == "user":
+                # Serialize User message as a prompt and save the assistant response as an output
+                assistant_response: Union[ChatCompletionMessage, None] = None
+                if i + 1 < len(conversation_data["messages"]):
+                    next_message = conversation_data["messages"][i + 1]
+                    if next_message["role"] == "assistant":
+                        assistant_response = next_message
+                        i += 1
+                new_prompt_name = f"{prompt_name}_{len(prompts) + 1}"
+
+                content = message["content"]
+                input = {}
+                # Message can contain text and multiple images
+                for val in content:
+                    if val["type"] == "text":
+                        input["data"] = val["text"]
+                    if val["type"] == "image_url":
+                        if input.get("attachments") is None:
+                            input["attachments"] = []
+                        image_url: str = val["image_url"]["url"]
+                        attachment_kind = (
+                            "base64"
+                            if image_url.startswith("data:")
+                            else "file_uri"
+                        )
+                        attachment_data = AttachmentDataWithStringValue(
+                            kind=attachment_kind, value=image_url
+                        )
+                        input["attachments"].append(
+                            Attachment(
+                                data=attachment_data, mime_type="image/*"
```

Comment:
So there's no way of detecting mime type from openAI? If it's a file_uri can't we read the extension?

Reply:
file_uri is basically any non-base64 uri -- in practice it's usually a url to some online blob, which won't necessarily have any extension. I *think* we could technically use a regex to look for extension in base64 case, if it's in format `data:image/x` so I can try that

Reply:
Not a blocker, if it takes more than 10 mins we can create a followup issue